core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -56,6 +56,14 @@
         <T extends AbstractManagedChannelImplBuilder<T>> extends ManagedChannelBuilder<T> {
   private static final String DIRECT_ADDRESS_SCHEME = "directaddress";
 
+  public static ManagedChannelBuilder<?> forAddress(String name, int port) {
+    throw new UnsupportedOperationException("Subclass failed to hide static factory");
```

Comment:
Why don't we just make this throw in a normal "unsupported" message and let subclasses override when it is supported?

Reply:
Not sure what you mean, can you clarify?

Reply:
Don't blame the subclasses with "Subclass failed to hide static factory". Basically, have a message that is "okay" here and let subclasses override when they feel like it.

Although that would remove the need for the test (although a test like that could have found `ForwardingChannelBuilder` which doesn't extend from this class).

Reply:
Since this is an internal class, I think it would be an error.  All implementations are owned by us, and we do want to override.